content/transfer-fee.md

```diff
@@ -0,0 +1,959 @@
+ - Withheld tokens can be withdrawn directly from the recipient accounts or can be harvested back to the mint and then withdrawn
+ - Transfers with mints using the `transfer fee` extension need to use the `transferCheckedWithFee` instruction
+
+# Overview
+
+Suppose you're a Solana game developer and you're making a large open world multiplayer role playing game. You'll have a currency in this game that all the players will earn and trade with. To make the economy in the game circular, you may want to charge a small transfer fee every time this currency changes hands, you'd call this the developer tax. This can be accomplished with the `transfer fee` extension. The neat part is this will work on every transfer, in-game and out! 
+
+The Token Extension Program's `transfer fee` extension enables you to configure a transfer fee on a mint such that fees are assessed at the protocol level. On every transfer, some amount of that mint is withheld on the recipient account which cannot be used by the recipient. At any point after the transfer, the `withdraw` authority can claim these withheld tokens.
+
+The `transfer fee` extension is customizable and updatable. Here are the inputs that we'll delve into a bit later:
+ - Fee basis points: This is the fee assessed on every transfer. For example, if 1000 tokens with 50 basis points are transferred, it will yield 5 tokens.
+ - Maximum fee: The cap on transfer fees. With a maximum fee of 5000 tokens, a transfer of 10,000,000,000,000 tokens will only yield 5000 tokens.
+ - Transfer fee authority: The entity that can modify the fees.
+ - Withdraw withheld authority: The entity that can move tokens withheld on the mint or token accounts.
+
+## Calculating fee basis points
+
+Before we go into the extension, here's a quick intro to "fee basis points".
+
+A basis point is a unit of measurement used in finance to describe the percentage change in the value or rate of a financial instrument. One basis point is equivalent to 0.01% or 0.0001 in decimal form.
+
+To get the fee we must calculate it as follows:
+
+$$ Fee = {token_amount * fee_basis_points \over 10000} $$
+
+The constant 10,000 is used to convert the fee basis point percentage to the equivalent amount.
+
+## Configuring a mint with a transfer fee
+
+Initializing a mint with the `transfer fee` extension involves three instructions:
+- `SystemProgram.createAccount`
+- `createInitializeTransferFeeConfigInstruction`
+- `createInitializeMintInstruction`
+
+The first instruction `SystemProgram.createAccount` allocates space on the blockchain for the mint account. This instruction accomplishes three things:
+ - Allocates `space`
+ - Transfers `lamports` for rent
+ - Assigns to it's owning program
+
+As with all Token Extension Program's mints, we need to calculate the space and lamports needed for the mint. We can get these by calling `getMintLen` and `getMinimumBalanceForRentExemption`
+
+```ts
+const extensions = [ExtensionType.TransferFeeConfig]
```

Comment:
Remember to format these too per CONTRiBUTING.md (just paste them into a file and save it, let prettier do it's work, then paste back).